app/views/spree/admin/orders/_form.html.haml

```diff
@@ -1,68 +1,35 @@
-          %col{style: "width: 8%;"}/
-        %thead#line-items
-          %tr
-            %th
-              = t(:item_description)
-            %th.price
-              = t(:price)
-            %th.qty
-              = t(:qty)
-            %th.total
-              %span
-                = t(:total)
-            %th.orders-actions.actions
-        %tbody
-          = f.fields_for :line_items do |li_form|
-            = render partial: 'spree/admin/orders/line_item', locals: { f: li_form }
-        %tbody#subtotal.no-border-top
-          %tr#subtotal-row
-            %td{colspan: "3"}
-              %b
-                = t(:subtotal)
-                \:
-            %td.total.align-center
-              %span
-                = @order.display_item_total.to_html
-            %td.actions
-        %tbody#order-charges.no-border-top
-          - @order.adjustments.eligible.each do |adjustment|
-            %tr
-              %td{colspan: "3"}
+    = render :partial => 'spree/shared/error_messages', :locals => { :target => @line_item }
+
+  = render :partial => "spree/admin/orders/shipment", :collection => @order.shipments, :locals => { :order => order }
+
+  - if order.adjustments.eligible.exists?
+    %fieldset.no-border-bottom
+      %legend= Spree.t('adjustments')
+      %table
+        %thead
+          %th= Spree.t('name')
+          %th= Spree.t('amount')
+        %tbody#order-charges.with-border{"data-hook" => "order_details_adjustments"}
+          - order.adjustments.eligible.each do |adjustment|
+            - next if ((adjustment.originator_type == 'Spree::TaxRate') and (adjustment.amount == 0)) || adjustment.originator_type == 'Spree::ShippingMethod'
+            %tr.total
+              %td
                 %strong
                   = adjustment.label
                   \:
               %td.total.align-center
-                %span= adjustment.display_amount.to_html
-              %td.actions
-        %tbody#order-total.grand-total.no-border-top
-          %tr
-            %td{colspan: "3"}
-              %b
-                = t(:order_total)
-                \:
-            %td.total.align-center
-              %span#order_total
-                = @order.display_total.to_html
-            %td.actions
+                %span= adjustment.display_amount
 
-      = render partial: 'spree/admin/orders/_form/distribution_fields'
+  - if order.line_items.exists?
+    %fieldset#order-total.no-border-bottom{"data-hook" => "order_details_total"}
+      %legend= Spree.t(:order_total)
+      %span.order-total= order.display_total
 
-      .filter-actions.actions{"ng-show" => "distributionChosen()"}
-        = button t(:update_and_recalculate_fees), 'icon-refresh'
-        %span.or
-          = t(:or)
-        = link_to_with_icon 'button icon-arrow-left', t(:back), admin_orders_url
   = javascript_tag do
-    = render partial: 'spree/admin/shared/update_order_state', handlers: [:js]
+    var order_number = '#{@order.number}';
```

Comment:
@luisramos0 this is the JS tag that doesn't work with Haml. L31 works fine, but from L32 to 34, Ruby code is not taken into account. 

The JS error on saving the order comes from https://github.com/openfoodfoundation/spree/blob/f55722b38db7e706a8521c9091a0e00119bb4d20/backend/app/assets/javascripts/admin/variant_autocomplete.js.erb#L69 which fails with `shipments is not defined`.